wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,202 @@
+unified APIs that the entire ecosystem can depend on, reducing the maintenance burden across
+projects and addressing scenarios that impede node drain or cause improper pod termination. Our
+objective is to create easily configurable, out-of-the-box solutions that seamlessly integrate with
+existing APIs and behaviors. We will strive to make these solutions minimalistic and extensible to
+support advanced use cases across the ecosystem.
+
+To properly solve the node drain, we must first understand the node lifecycle. This includes
+provisioning/sunsetting of the nodes, PodDisruptionBudgets, API-initiated eviction and node
+shutdown. This then impacts both the node and pod autoscaling, de/scheduling, load balancing, and
+the applications running in the cluster. All of these areas have issues and would benefit from a
+unified approach.
+
+### In scope
+
+- Explore a unified way of draining the nodes and managing node maintenance by introducing new APIs
+  and extending the current ones. This includes exploring extension to or interactions with the Node
+  object.
+- Analyze the node lifecycle, the Node API, and possible interactions. We want to explore augmenting
+  the Node API to expose additional state or status in order to coalesce other core Kubernetes and
+  community APIs around node lifecycle management.
+- Improve the disruption model that is currently implemented by API-initiated Eviction API and PDBs.
+  Improve the descheduling, availability and migration capabilities of today's application
+  workloads. Also explore the interactions with other eviction mechanisms.
+- Coordinate pod termination and issues around de/scheduling, preemption and eviction.
+- Improve the Graceful/Non-Graceful Node Shutdown and consider how this affects the node lifecycle.
+- Improve the scheduling and pod/node autoscaling to take into account ongoing node maintenance and
+  the new disruption model/evictions. This includes balancing of the pods according to scheduling
+  constraints. 
+- Consider improving the pod lifecycle of DaemonSets and static pods during a node maintenance.
+- Explore the cloud provider use cases and how they can hook into the node lifecycle. So that the
+  users can use the same APIs or configurations across the board.
+- Migrate users of the eviction based kubectl-like drain (kubectl, cluster autoscaler, karpenter,
+  ...) and other scenarios to use the new unified node draining approach.
+- Explore possible scenarios behind the reason why the node was terminated/drained/killed and how to
+  track and react to each of them. Consider past discussions/historical perspective
+  (e.g. "tombstones").
+- Explore feedback mechanism for ensuring schedulability (e.g. readiness) and capabilities of the 
+  node. These can apply during the provisioning of the node, but also during the rest of the
+  node lifecycle.
+
+### Out of scope
+
+- Implementing cloud provider specific logic, the goal is to have high-level API that the providers
```

Comment:
@atiratree it is important to. mention that Core APIs that are within kubernetes/kubernetes repo MUST have a reference implementation that allow to test these APIs to guarantee the semantics across the ecosystem.

The alternative is to follow a model like Gateway API where official CRDs with API review are released and follow an independent conformance process https://github.com/kubernetes-sigs/gateway-api

What will be the model of development of this high level APIs? can we be explicit about it?